xarray/core/dataarray.py

```diff
@@ -2295,19 +2295,19 @@ def from_dict(cls, d: dict) -> "DataArray":
         return obj
 
     @classmethod
-    def from_series(cls, series: pd.Series) -> "DataArray":
+    def from_series(cls, series: pd.Series, sparse: bool = False) -> "DataArray":
         """Convert a pandas.Series into an xarray.DataArray.
 
         If the series's index is a MultiIndex, it will be expanded into a
         tensor product of one-dimensional coordinates (filling in missing
         values with NaN). Thus this operation should be the inverse of the
         `to_series` method.
         """
-        # TODO: add a 'name' parameter
+        # TODO: add a 'name' parameter?
         name = series.name
         df = pd.DataFrame({name: series})
-        ds = Dataset.from_dataframe(df)
-        return ds[name]
+        ds = Dataset.from_dataframe(df, sparse=sparse)
+        return ds[name]  # type: ignore
```

Comment:
When a Series has no name, Series.name is None. There is a ton of logic in xarray that treats None as a special case. And while ``hash(None)`` works and ``isinstance(None, Hashable)`` returns True, mypy doesn't consider None as a valid input for Hashable. So this is bound to trigger a mypy failure as soon as annotations are added to pandas, and a ``Dataset.__getitem__`` failure as soon as its implementation for any reason starts handling None explicitly.

Could you please change to
```python
# Don't use series.name, in case it is None
df = pd.DataFrame({"__temp": series})
ds = Dataset.from_dataframe(df, sparse=sparse)
out = ds["__temp"]
out.name = series.name
return out
```
The  ``# TODO: add a 'name' parameter?`` can go away IMHO.

Reply:
Good catch. I switched to using the `_THIS_ARRAY` sentinel value which we already use elsewhere for this purpose.